File: src/main/java/com/company/erp/features/devices/dto/DeviceDto.java
package com.company.erp.features.devices.dto;

import com.company.erp.features.devices.entity.Device;
import com.company.erp.features.devices.entity.DevicePlatform;

import java.time.Instant;

public record DeviceDto(
        Long id,
        String deviceId,
        DevicePlatform platform,
        String appVersion,
        Instant updatedAt
) {
    public static DeviceDto from(Device d) {
        return new DeviceDto(d.getId(), d.getDeviceId(), d.getPlatform(),
                d.getAppVersion(), d.getUpdatedAt());
    }
}

File: app/Features/Devices/Dto/DeviceDto.php
<?php

namespace App\Features\Devices\Dto;

use App\Features\Devices\Models\Device;

/** Port of the Spring DeviceDto record. */
final class DeviceDto
{
    /**
     * @return array<string, mixed>
     */
    public static function from(Device $d): array
    {
        return [
            'id' => (int) $d->id,
            'deviceId' => $d->device_id,
            'platform' => $d->platform,
            'appVersion' => $d->app_version,
            'updatedAt' => $d->updated_at ? $d->updated_at->utc()->format('Y-m-d\TH:i:s\Z') : null,
        ];
    }
}
